src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,114 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+
+	"github.com/pachyderm/pachyderm/v2/src/admin"
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/debug"
+	"github.com/pachyderm/pachyderm/v2/src/enterprise"
+	"github.com/pachyderm/pachyderm/v2/src/identity"
+	"github.com/pachyderm/pachyderm/v2/src/license"
+	"github.com/pachyderm/pachyderm/v2/src/proxy"
+	"github.com/pachyderm/pachyderm/v2/src/transaction"
+	"github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	"github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewMux(pachClientFactory func(context.Context) *client.APIClient) http.Handler {
```

Comment:
Can the grpc.ClientConn be passed here directly, or be returned by the closure?

If we keep this context -> client.APIClient thing then we need to use `internal/pctx` so we have logging.  No `context.Background()`

Reply:
@zmajeed This will be relevant to your PR 👆